src/OrleansRuntime/Streams/QueueBalancer/StreamQueueBalancerFactory.cs

```diff
@@ -21,7 +22,7 @@ internal class StreamQueueBalancerFactory
         /// <param name="siloMaturityPeriod">Maturity Period of a silo for queue rebalancing purposes</param>
         /// <returns>Constructed stream queue balancer</returns>
         public static IStreamQueueBalancer Create(
-            StreamQueueBalancerType balancerType,
+            string balancerType,
             string strProviderName,
             ISiloStatusOracle siloStatusOracle,
```

Comment:
My thinking here is to avoid passing anything that can be obtained by DI, passing only the things that the queue balancer manager alone has, like balancer type, provider name, and maturity period.  The container can be obtained from DI, right?